website/docs/r/kubernetes_cluster_node_pool.html.markdown

```diff
@@ -96,6 +96,8 @@ The following arguments are supported:
 
 ~> **Note:** FIPS support is in Public Preview - more information and details on how to opt into the Preview can be found in [this article](https://docs.microsoft.com/azure/aks/use-multiple-node-pools#add-a-fips-enabled-node-pool-preview).
 
+* `gateway_profile` - (Optional) A `gateway_profile` block as defined below. Only valid when `mode` is set to `Gateway`.
```

Comment:
For "Only valid when `mode` is set to `Gateway`", could you describe it as a `Note` and add a validation in `CutomizeDiff` method?
```suggestion
* `gateway_profile` - (Optional) A `gateway_profile` block as defined below.
```